### PR TITLE
refactor: files structure in `core` folder

### PR DESCRIPTION
**Pull request by artmakarov**
Description:

--

QA-test:

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.23.0-pr-463-ebc0-ebc05615.zip

### CODE REVIEW

File: client-app/pages/catalog.vue [hunk 1]
```diff
@@ -325,11 +325,10 @@ import { useRoute } from "vue-router";
 import {
   buildBreadcrumbs,
   DEFAULT_PAGE_SIZE,
-  FacetItem,
-  FacetValueItem,
+  FacetItemType,
+  FacetValueItemType,
```
Comment (goldenmaya) [inline client-app/pages/catalog.vue:329]: I thought we gonna use separated imports like in [quotes.vue](https://github.com/VirtoCommerce/vc-theme-b2b-vue/pull/463/files/1f935acc008a76c0b1ff33a4b61227d473627a45#diff-2c685f40a20748d414ae1dad4ecc25a92b60a6de7bde8a06d59b68a4c95c678dR161)

File: client-app/pages/account/addresses.vue [hunk 2]
```diff
@@ -230,7 +230,8 @@ import { clone } from "lodash";
 import { computed, ComputedRef, onMounted, Ref, ref } from "vue";
 import { useI18n } from "vue-i18n";
 import { useCountries, usePageHead } from "@/core/composables";
-import { AddressType, ISortInfo } from "@/core/types";
+import { AddressType } from "@/core/enums";
+import { ISortInfo } from "@/core/types";
 import { useUserAddresses } from "@/shared/account";
 import { BackButtonInHeader } from "@/shared/layout";
 import { MemberAddressType } from "@/xapi/types";
```
Comment (goldenmaya) [inline client-app/pages/account/addresses.vue:237]: import type

File: client-app/pages/company/info.vue [hunk 3]
```diff
@@ -237,7 +237,9 @@ import { useField } from "vee-validate";
 import { computed, ref, watch } from "vue";
 import { useI18n } from "vue-i18n";
 import * as yup from "yup";
-import { AddressType, ISortInfo, usePageHead, XApiPermissions } from "@/core";
+import { usePageHead } from "@/core/composables";
+import { AddressType, XApiPermissions } from "@/core/enums";
+import { ISortInfo } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/pages/company/info.vue:242]: import type

File: client-app/pages/catalog.vue (hunk 1 above)
Comment (artmakarov) [reply client-app/pages/catalog.vue:329]: This PR refers to the "core" folder.  Next we will make PRs for other folders.

File: client-app/pages/account/quotes.vue [hunk 4]
```diff
@@ -158,8 +158,10 @@ import { useBreakpoints, breakpointsTailwind } from "@vueuse/core";
 import { computed, ref, shallowRef, watch } from "vue";
 import { useI18n } from "vue-i18n";
 import { useRouter } from "vue-router";
-import { getSortingExpression, ISortInfo, QueryParamName, getSortInfoFromStringExpression } from "@/core";
 import { useElementVisibility, useRouteQueryParam, usePageHead } from "@/core/composables";
+import { QueryParamName } from "@/core/enums";
+import type { ISortInfo } from "@/core/types";
+import { getSortingExpression, getSortInfoFromStringExpression } from "@/core/utilities";
 import { PageToolbarBlock, useUserQuotes } from "@/shared/account";
 import { QuoteType } from "@/xapi/types";
```
Comment (goldenmaya) [inline client-app/pages/account/quotes.vue:166]: import type

File: client-app/pages/company/members.vue [hunk 5]
```diff
@@ -374,8 +374,8 @@ import { breakpointsTailwind, computedEager, onClickOutside, useBreakpoints } fr
 import { computed, onMounted, ref, shallowRef } from "vue";
 import { useI18n } from "vue-i18n";
 import { useElementVisibility, usePageHead } from "@/core/composables";
-import { XApiPermissions } from "@/core/constants";
-import { FacetItem, FacetValueItem, ISortInfo } from "@/core/types";
+import { XApiPermissions } from "@/core/enums";
+import { FacetItemType, FacetValueItemType, ISortInfo } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/pages/company/members.vue:378]: import type

File: client-app/shared/catalog/components/products-filters.vue [hunk 6]
```diff
@@ -87,7 +87,7 @@
 import { eagerComputed, useBreakpoints, breakpointsTailwind } from "@vueuse/core";
 import _ from "lodash";
 import { watch, PropType, ref, shallowReactive } from "vue";
-import { FacetItem } from "@/core/types";
+import { FacetItemType } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/shared/catalog/components/products-filters.vue:90]: import type

File: client-app/shared/catalog/composables/useProducts.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { computed, inject, readonly, ref, shallowRef } from "vue";
-import { configInjectionKey } from "@/core/constants";
-import { FacetItem } from "@/core/types";
+import { configInjectionKey } from "@/core/injection-keys";
+import { FacetItemType } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/shared/catalog/composables/useProducts.ts:3]: import type

File: client-app/shared/catalog/composables/useProducts.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
...
 import { Logger, rangeFacetToCommonFacet, termFacetToCommonFacet } from "@/core/utilities";
 import { searchProducts } from "@/xapi/graphql/catalog";
 import { Product } from "@/xapi/types";
```
Comment (goldenmaya) [inline client-app/shared/catalog/composables/useProducts.ts:6]: import type

File: client-app/shared/catalog/types/search.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { FacetItem } from "@/core/types";
+import { FacetItemType } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/shared/catalog/types/search.ts:1]: import type

File: client-app/shared/company/components/filter-facet.vue [hunk 10]
```diff
@@ -17,20 +17,20 @@
 <script setup lang="ts">
 import _ from "lodash";
 import { PropType, ref, watch } from "vue";
-import { FacetItem } from "@/core/types";
+import { FacetItemType } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/shared/company/components/filter-facet.vue:20]: import type

File: client-app/shared/company/composables/useOrganizationContactsFilterFacets.ts [hunk 11]
```diff
@@ -8,14 +8,14 @@ import {
   PURCHASING_AGENT,
   STORE_ADMINISTRATOR,
   STORE_MANAGER,
-} from "@/core/constants";
-import { FacetItem, FacetValueItem } from "@/core/types";
+} from "@/core/security-constants";
+import { FacetItemType, FacetValueItemType } from "@/core/types";
```
Comment (goldenmaya) [inline client-app/shared/company/composables/useOrganizationContactsFilterFacets.ts:12]: import type

**Review by goldenmaya — APPROVED** [review]

**Pull request merged**